egs/iam/s5/local/unk_arc_post_to_transcription.py

```diff
@@ -10,42 +10,40 @@
           phone and phoneID, Eg. a 217, phoneID of 'a' is 217. 
   words: File name of a file that contains the words.txt, (symbol-table for words). 
          word and wordID. Eg. ACCOUNTANCY 234, wordID of 'ACCOUNTANCY' is 234.
-  unk: ID of <unk>. Eg. 231
-  input-ark: file containing hypothesis transcription with <unk>
-  out-ark: file containing hypothesis transcription without <unk>
+  unk: ID of <unk>. Eg. 231.
+  bestarcpost: A file in arc-post format, which is a list of timing info and posterior 
+               of arcs along the one-best path from the lattice.
+               E.g. 506_m01-049-00 8 12  1 7722  282 272 288 231
+                    <utterance-id> <start-frame> <num-frames> <posterior> <word> [<ali>] [<phone1> <phone2>...]
+  output-text: File containing hypothesis transcription with <unk> recognized by the unk-model.
+               E.g. A move to stop mr. gaitskell.
   
   Eg. local/unk_arc_post_to_transcription.py lang/phones.txt lang/words.txt data/lang/oov.int
 
-  Eg. hypothesis transcription: A move to <unk> mr. gaitskell.
-      converted transcription: A move to stop mr. gaitskell.
-
 """
-
 import argparse
 import os
 import sys
-import numpy as np
-from scipy import misc
 parser = argparse.ArgumentParser(description="""uses phones to convert unk to word""")
 parser.add_argument('phones', type=str, help='File name of a file that contains the symbol-table for phones. Each line must be: <phone> <phoneID>')
 parser.add_argument('words', type=str, help='File name of a file that contains the symbol-table for words. Each line must be: <word> <word-id>')
 parser.add_argument('unk', type=str, default='-', help='File name of a file that contains the ID of <unk>. The content must be: <oov-id>, e.g. 231')
-parser.add_argument('--input-ark', type=str, default='-', help='where to read the input data')
-parser.add_argument('--out-ark', type=str, default='-', help='where to write the output data')
+parser.add_argument('--bestarcpost', type=str, default='-', help='A file in arc-post format, which is a list of timing info and posterior of arcs along the one-best path from the lattice')
+parser.add_argument('--output-text', type=str, default='-', help='File containing hypothesis transcription with <unk> recognized by the unk-model')
 args = parser.parse_args()
 
 ### main ###
-phone_fh = open(args.phones, 'r')
+phone_fh = open(args.phones, 'r') #create file handles 
 word_fh = open(args.words, 'r')
 unk_fh = open(args.unk,'r')
-if args.input_ark == '-':
+if args.bestarcpost == '-':
     input_fh = sys.stdin
 else:
-    input_fh = open(args.input_ark,'r')
-if args.out_ark == '-':
+    input_fh = open(args.bestarcpost,'r')
+if args.output_text == '-':
     out_fh = sys.stdout
 else:
-    out_fh = open(args.out_ark,'wb')
+    out_fh = open(args.output_text,'wb')
 
 phone_dict = dict() #stores mapping from phoneID(int) to phone(char)
 phone_data_vect = phone_fh.read().strip().split("\n")
```

Comment:
phone_dict => id2phone, phone_data_vect => phones_table (or phones_data)
#stores mapping from phoneID(int) to phone(char) => # Stores the mapping from phone_id (int) to phone (char)